runtime/parachains/src/scheduler/tests.rs

```diff
@@ -18,26 +18,30 @@ use super::*;
 
 use frame_support::assert_ok;
 use keyring::Sr25519Keyring;
-use primitives::{BlockNumber, CollatorId, SessionIndex, ValidatorId};
+use primitives::{BlockNumber, CollatorId, SessionIndex, ValidationCode, ValidatorId};
 
 use crate::{
 	configuration::HostConfiguration,
 	initializer::SessionChangeNotification,
 	mock::{
-		new_test_ext, Configuration, MockGenesisConfig, Paras, ParasShared, Scheduler, System, Test,
+		new_test_ext, Configuration, MockGenesisConfig, Paras, ParasShared, RuntimeOrigin,
+		Scheduler, System, Test,
 	},
 	paras::{ParaGenesisArgs, ParaKind},
 };
 
 fn schedule_blank_para(id: ParaId, parakind: ParaKind) {
+	let validation_code: ValidationCode = vec![1, 2, 3].into();
 	assert_ok!(Paras::schedule_para_initialize(
 		id,
 		ParaGenesisArgs {
 			genesis_head: Vec::new().into(),
-			validation_code: vec![1, 2, 3].into(),
+			validation_code: validation_code.clone(),
 			para_kind: parakind,
 		}
 	));
+
+	assert_ok!(Paras::add_trusted_validation_code(RuntimeOrigin::root(), validation_code));
```

Comment:
Note that `add_trusted_validation_code` is used outside of `paras` tests to bypass prechecking. Mainly because most pallets don't setup active validators, but prechecking does require at least a single vote to get concluded.
Hence we avoid a lot of boilerplate unrelated to hrmp and scheduler tests.

The method itself, however, was introduced by https://github.com/paritytech/polkadot/issues/4575 and doc comments states
```rust
/// ...
/// This function is mainly meant to be used for upgrading parachains that do not follow
/// the go-ahead signal while the PVF pre-checking feature is enabled.
```

However, at this point everyone uses https://github.com/paritytech/cumulus/pull/517 and this method can be put under `#[cfg(test)]`. If this is true we can create a follow-up to clean paras from dead code.